### PR TITLE
Update of Documentation 'ESP Matter light' for 3.1 release

### PR DESCRIPTION
**Pull request by lboue**
### Checklist
1. [x] Please provide specific title of the PR describing the change, including the component name (*eg. „Update of Documentation link on Readme.md“*)
2. [x] Please provide related links (*eg. Issue which will be closed by this Pull Request*)
3. [x] Please **update relevant Documentation** if applicable
4. [x] Please check [Contributing guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/contributing.html)
5. [x] Please **confirm option to "Allow edits and access to secrets by maintainers"** when opening a Pull Request

*This entire section above can be deleted if all items are checked.*

-----------
## Description of Change
Update doc for 3.1 release.

## Tests scenarios
N/A

## Related links
Related PR: #10202

### CODE REVIEW

File: idf_component_examples/esp_matter_light/README.md [hunk 1]
```diff
@@ -74,17 +74,24 @@ Those are the tested devices that have a WS2812 RGB LED and can run BLE, Wi-Fi a
 In case it is necessary to change the Button Pin or the REG LED Pin, please use the `menuconfig`
 `idf.py menuconfig` and change the Menu Option `Light Matter Accessory`
 
-## Using OpenThread with Matter
+## Building the Application using OpenThread and Matter
 
 This is possible with the ESP32-C6.  
-It is neessasy to have a Thread Border Routed in the Matter Environment. Check you matter hardware provider.  
+It is necessary to have a Thread Border Router in the Matter Environment. Check you Matter hardware provider.
```
Comment (SuGlider) [inline idf_component_examples/esp_matter_light/README.md:90]:
```suggestion
It is necessary to have a Thread Border Router in the Matter Environment. Check your Matter hardware provider.  
```

**Comment by SuGlider** [comment]
@lboue - Thanks for the PR!
I have added a few more changes.
Please let me know.

**Comment by SuGlider** [comment]
@lboue - I think that it shoud target master because the changed  will be also merged into 3.1, automaticaly.
Is it right, @me-no-dev?

**Comment by me-no-dev** [comment]
>  I think that it shoud target master because the changed will be also merged into 3.1, automaticaly.

yes. that is correct. Now they will be fixed only for 3.1

**Pull request closed without merging**